docs/resources/instancev2.md

```diff
@@ -0,0 +1,524 @@
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+}
+
+resource "gcore_instancev2" "instance" {
+  flavor_id     = "g1w-standard-4-8"
+  name          = "my-windows-instance"
+  password      = "my-s3cR3tP@ssw0rd"
+
+  volume {
+    source     = "existing-volume"
+    volume_id  = gcore_volume.boot_volume_windows.id
+    boot_index = 0
+  }
+
+  interface {
+    type = "external"
+    name = "my-external-interface"
+  }
+
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+}
+```
+
+### Advanced examples
+
+#### Creating instance with floating ip
+
+```terraform
+resource "gcore_reservedfixedip" "fixed_ip" {
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+  type       = "subnet"
+  network_id = gcore_network.network.id
+  subnet_id  = gcore_subnet.subnet.id
+}
+
+resource "gcore_floatingip" "floating_ip" {
+  project_id       = data.gcore_project.project.id
+  region_id        = data.gcore_region.region.id
+  fixed_ip_address = gcore_reservedfixedip.fixed_ip.fixed_ip_address
+  port_id          = gcore_reservedfixedip.fixed_ip.port_id
+}
+
+resource "gcore_instancev2" "instance" {
+  flavor_id     = "g1-standard-2-4"
+  name          = "my-instance"
+  keypair_name  = "my-keypair"
+
+  volume {
+    source     = "existing-volume"
+    volume_id  = gcore_volume.boot_volume.id
+    boot_index = 0
+  }
+
+  interface {
+    type    = "reserved_fixed_ip"
+    name    = "my-floating-ip-interface"
+    port_id = gcore_reservedfixedip.fixed_ip.port_id
+
+    fip_source      = "existing"
+    existing_fip_id = gcore_floatingip.floating_ip.id
+  }
+
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+}
+```
+
+#### Creating instance with a reserved public interface
+
+```terraform
+resource "gcore_reservedfixedip" "fixed_ip" {
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+  type       = "external"
+}
+
+resource "gcore_instancev2" "instance" {
+  flavor_id     = "g1-standard-2-4"
+  name          = "my-instance"
+  keypair_name  = "my-keypair"
+
+  volume {
+    source     = "existing-volume"
+    volume_id  = gcore_volume.boot_volume.id
+    boot_index = 0
+  }
+
+  interface {
+    type    = "reserved_fixed_ip"
+    name    = "my-reserved-public-interface"
+    port_id = gcore_reservedfixedip.fixed_ip.port_id
+  }
+
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+}
+```
+
+
+#### Creating instance with custom security group
+
+This example demonstrates how to create an instance with a custom security group. The security group allows all
+incoming traffic on ports 22, 80, and 443. Outgoing traffic is allowed on all ports, except port 25 for security reasons.
+
+
+```terraform
+resource "gcore_securitygroup" "web_server_security_group" {
+  name       = "web server only"
```

Comment:
I'm thinking about this naming: shouldn't it be my-security-group or my-web-server-security-group for consistency with other namings?